src/services/doppler-api-client.ts

```diff
@@ -0,0 +1,78 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession, DopplerAPIConnectionData } from './app-session';
+import { RefObject } from 'react';
+import { SubscriberList, SubscriberListState } from './shopify-client';
+
+export interface DopplerAPIClient {
+  getListData(idList: number, apikey: string): Promise<ResultWithoutExpectedErrors<SubscriberList>>;
+}
+
+export class HttpDopplerAPIClient implements DopplerAPIClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerAPIConnectionData(): DopplerAPIConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler API connection data is not available');
+    }
+    return { jwtToken: connectionData.jwtToken, userAccount: connectionData.userData.user.email };
+  }
+
+  private mapList(data: any): SubscriberList {
+    return {
+      name: data.name,
+      id: data.listId,
+      amountSubscribers: data.subscribersCount,
+      state:
+        data.currentStatus === SubscriberListState.ready
+          ? SubscriberListState.ready
+          : SubscriberListState.synchronizingContacts,
+    };
+  }
+
+  public async getListData(
+    listId: number,
+    apikey: string,
+  ): Promise<ResultWithoutExpectedErrors<SubscriberList>> {
+    try {
+      // eslint-disable-next-line
```

Comment:
Is it really necessary?

Reply:
To avoid `jwtToken` warning of not used variable

Reply:
Thanks for the explanation, maybe you could add it to the comment.

Another option is that since it is a temporal code, you could keep the warning.

Another one could be avoiding deconstructing that value while you do not need it.